gateway-js/src/outOfBandReporter.ts

```diff
@@ -63,32 +79,34 @@ export class OutOfBandReporter {
 
     const responseBody: string | undefined = await response?.text();
 
-    const oobVariables: ApiMonitoringReport = {
-      error: {
-        code: errorCode,
-        message: error.message ?? error
-      },
-      request: {
-        url: request.url,
-        body: await request.text()
+    const variables: OobReportMutationVariables = {
+      input: {
+        error: {
+          code: errorCode,
+          message: error.message ?? error,
```

Comment:
@trevor-scheer is this a typo? I was attempting to update our version of `@apollo/gateway` from `0.24` to `0.31` and now I am getting a typescript error

```
$ tsc
node_modules/@apollo/gateway/src/outOfBandReporter.ts:86:11 - error TS2322: Type 'string | Error' is not assignable to type 'string'.
  Type 'Error' is not assignable to type 'string'.

86           message: error.message ?? error,
             ~~~~~~~

  node_modules/@apollo/gateway/src/__generated__/graphqlTypes.ts:28:3
    28   message?: Maybe<Scalars['String']>;
         ~~~~~~~
    The expected type comes from property 'message' which is declared here on type 'Error'
```

Reply:
It looks like the error actually first occurs in version `^0.30`

Reply:
Thanks @smyrick, taking a look now. Agree the types are off, surprised we're not seeing the error on our end as well 🤔

Reply:
Possibly a different TS version or config?

Ours recognizes `error.message` as a `string`, which means we'll (theoretically) never evaluate the `??`. cc @Y-Guo, I'm curious if the `?? error` is serving a runtime purpose and maybe our types are off further up (i.e. maybe `error` isn't an `Error` for sure?)

Reply:
We are running TS with `4.3.4`

```
  "compilerOptions": {
    "allowJs": true,
    "baseUrl": ".",
    "checkJs": true,
    "declaration": true,
    "declarationMap": true,
    "experimentalDecorators": true,
    "incremental": true,
    "lib": ["esnext"],
    "module": "commonjs",
    "noUnusedLocals": true,
    "esModuleInterop": true,
    "outDir": "dist",
    "sourceMap": true,
    "target": "es2020",
    "tsBuildInfoFile": ".tsbuildinfo"
  },
```